### PR TITLE
use snappy compression for cloning instead of gzip

### PR DESCRIPTION
**Pull request by mhenriks**
Signed-off-by: Michael Henriksen <mhenriks@redhat.com>

**What this PR does / why we need it**:

Snappy is optimized for high throughput and performs much better than gzip for our use case

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Use Snappy compression for network cloning
```

### CODE REVIEW

File: pkg/uploadserver/uploadserver.go [hunk 1]
```diff
@@ -457,14 +460,21 @@ func filesystemCloneProcessor(stream io.ReadCloser, destDir string) error {
...
-	if err = util.UnArchiveTar(gzr, destDir); err != nil {
+	if err := util.UnArchiveTar(newSnappyReadCloser(stream), destDir); err != nil {
 		return errors.Wrapf(err, "error unarchiving to %s", destDir)
 	}
 
 	return nil
 }
+
+func newContentReader(stream io.ReadCloser, contentType string) io.ReadCloser {
+	if contentType == BlockdeviceClone {
```
Comment (maya-r) [inline pkg/uploadserver/uploadserver.go:471]: I'm finding it strange that we need this conditional but the old code didn't

Comment (mhenriks) [reply pkg/uploadserver/uploadserver.go:471]: The data processor code can handle gzipped streams but not snappy compressed streams

**Review by awels — APPROVED** [review]
/lgtm
/approve
dang you reworked a lot of the cloner tests.

**Pull request merged**